src/luarocks/fs/lua.lua

```diff
@@ -544,86 +544,183 @@ end
 if socket_ok then
 
 local ltn12 = require("ltn12")
+local http = require("socket.http")
 local luasec_ok, https = pcall(require, "ssl.https")
 
-local redirect_protocols = {
-   http = http,
-   https = luasec_ok and https,
-}
+-- Patch ltn12 source used by LuaSocket to get fixed length
```

Comment:
> What do you mean by being not responsive? It updates too rarely?

Yes, in my experience, with the dots in step(), most of the delay is caused by the initial setup of the connection, the actual downloads of the payload run fast/are short, so there's only one or two `step` updates after the initial delay for typical small rocks. My copas-based spinner was more meant as a psychological "hang in there" than as a way to measure progress.

But well, since I committed the dots at one point I suppose there were situations where they were useful.